### DISCUSSION TITLE
Error when running executable?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [pyinstaller/pyinstaller#7220] [Help]
Hello - i have the following code - which works fine on Windows / MacOs when running in the terminal

```
import trafilatura as traf
if __name__ == '__main__':
  url = traf.fetch_url("http://www.goodfoodgroup.com")
  workText = traf.extract(url)    
  print(workText)
```

But when i create an executeable using pyinstaller (v 5.6.2) and start the program. I get the following error-message:
(on MacOs and on Windows)

```
PolziMacs-Mini:~ polzimac$ /Users/polzimac/Documents/DEV/Python-Diverses/Trafilatura/exmpl2 ; exit;
Traceback (most recent call last):
  File "configparser.py", line 789, in get
  File "collections/__init__.py", line 941, in __getitem__
  File "collections/__init__.py", line 933, in __missing__
KeyError: 'min_file_size'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "exmpl2.py", line 4, in <module>
  File "trafilatura/downloads.py", line 156, in fetch_url
  File "trafilatura/downloads.py", line 124, in _handle_response
  File "configparser.py", line 818, in getint
  File "configparser.py", line 808, in _get_conv
  File "configparser.py", line 803, in _get
  File "configparser.py", line 792, in get
configparser.NoOptionError: No option 'min_file_size' in section: 'DEFAULT'
[32845] Failed to execute script 'exmpl2' due to unhandled exception!
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
```

Why is this not working when i created an executeable with pyinstaller?

**Comment by bwoodsend** [comment]
trafilatura contains a settings file (`.../site-packages/trafilatura/settings.cfg`). You just need to add that file using `--collect-datas=trafilatura`. I found I also have to add `--hiddenimport=charset_normalizer.md__mypyc` to get rid of a `ModuleNotFoundError: No module named 'charset_normalizer.md__mypyc'` later on.

**Reply by webghostx** [reply]
I have the same problem, but with my own script. I have two instances of Configparser. It works with just one, the .ini from the second instance is ignored. how do I write the collect  into the .spec? I don't understand it in the documentation.

**Reply by bwoodsend** [reply]
I don't have anything to say on collecting data files beyond what's already in the documentation for [adding data files](https://pyinstaller.org/en/stable/usage.html?highlight=add-data#cmdoption-add-data) and for [locating them at runtime](https://pyinstaller.org/en/stable/runtime-information.html#using-file).

**Comment by Rapid1898-code** [comment]
Thanks a lot with this command everythign seems to work fine now with the exe:
`pyinstaller --onefile --collect-datas=trafilatura xyz.py`
